homework_2/internal/domain/services/feedProcessorService.go

```diff
@@ -4,37 +4,51 @@ import (
 	"context"
 	"log"
 
+	"github.com/pkg/errors"
+
 	"code-cadets-2021/homework_2/internal/domain/models"
 )
 
 type FeedProcessorService struct {
-	feed  Feed
+	feeds  []Feed
 	queue Queue
 }
 
 func NewFeedProcessorService(
-	feed Feed,
+	feeds []Feed,
 	queue Queue,
 ) *FeedProcessorService {
 	return &FeedProcessorService{
-		feed:  feed,
+		feeds:  feeds,
 		queue: queue,
 	}
 }
 
 func (f *FeedProcessorService) Start(ctx context.Context) error {
-	updates := f.feed.GetUpdates()
 	source := f.queue.GetSource()
+	var updates []chan models.Odd
+
+	for _, feed := range f.feeds {
+		updates = append(updates, feed.GetUpdates())
+	}
 
 	defer close(source)
 	defer log.Printf("shutting down %s", f)
 
-	for update := range updates {
-		update.Coefficient *= 2
-		source <- update
+	counter := 0
+	for {
+		select {
+			case msg, ok := <- updates[counter % len(updates)]:
+				if !ok {
+					return nil
+				}
+				msg.Coefficient *= 2
+				source <- msg
+		}
+		counter += 1
```

Comment:
Unless there is a reason why feeds need to be consumed in a synchronous fashion (A, B, A, B, A...), this only slows down the processing (as it waits for one feed to provide a result while the other might already have one available) and doesn't use the full potential of concurrency.

I suggest, instead of consuming alternately from both feeds, consume from them at the same time (whichever comes first) - take a look at this example - https://gobyexample.com/select

One more thing, you might want to consume from the context.Done() to see if the context is finished (if that falls in line with how the program should be working) - see example here - https://gobyexample.com/context